test/language/expressions/async-generators/return-suspendedYield-try-finally-throw.js

```diff
@@ -0,0 +1,44 @@
+// Copyright 2017 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Caitlin Potter <caitp@igalia.com>
+esid: pending
+description: >
+  AsyncGeneratorResumeNext:
+  If completion.[[Type]] is return, and generator.[[AsyncGeneratorState]] is
+  "suspendedYield", and generator is resumed within a try-block with an
+  associated finally block, resume execution within finally.
+flags: [async]
+---*/
+
+class Err extends Error {}
+var g = async function*() {
+  try {
+    yield 1;
+    $ERROR('Generator must be resumed in finally block.');
+  } finally {
+    throw new Err;
+    $ERROR('Generator must not be resumed.');
+  }
+};
+
+var it = g();
+it.next().then(function(ret) {
+  // Initial yield
+  assert.sameValue(ret.value, 1);
+  assert.sameValue(ret.done, false);
+
+  it.return('sent-value').then($DONE, function(err) {
+    // Throw in finally block
+    assert(err instanceof Err);
```

Comment:
if you store the reference for `new Error` you can throw it in the finally block and assert `err` is the same reference, right?

It also saves the need for an Error subclass